### PR TITLE
Accommodate incorrect validity map values

### PR DESCRIPTION
**Pull request by aaronwolen**
This supersedes #93 and accommodates the new legacy validity mode added in tiledb-inc/tiledb-r#517.

A new metadata tag, `soma_legacy_validity`, is now attached to all arrays created by `SOMA` objects. By default this value is `"false"` unless the TileDB-R legacy validity mode was enabled at creation time (i.e., `r.legacy_validity_mode`). When reading arrays from disk, the `AnnotationDataFrame` class will check for this tag on initialization and when performing reads or writes. If the tag is present and set to `"true"`, legacy validity mode is enabled globally (as it's not possible to set on a per-array basis). Legacy validity mode is also enabled when reading `AnnotationDataFrame` arrays that lack the tag, as this indicates the array was created with an older version of the package. These checks are limited to `AnnotationDataFrame` arrays because the incorrect validity map values only affect nullable string attributes. See TileDB-R's [release notes](https://github.com/TileDB-Inc/TileDB-R/blob/master/NEWS.md#breaking-changes) for more information.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -56,3 +56,4 @@ Suggests:
     SingleCellExperiment
 VignetteBuilder: knitr
 Config/testthat/edition: 2
+Additional_repositories: https://tiledb-inc.r-universe.dev
```
Comment (eddelbuettel) [inline DESCRIPTION:59]: Do we need this?

File: R/utils-tiledb.R [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+tiledb_ctx_set_key <- function(key, value) {
+  cfg <- tiledb::config(tiledb::tiledb_get_context())
+  cfg[key] <- value
+  invisible(tiledb::tiledb_ctx(config = cfg))
+}
+
+#' Toggle legacy validity mode in the global context
+#'
+#' Legacy validity mode is enabled if:
+#' - the array does not contain the metadata tag `"soma_legacy_validity"`
```
Comment (eddelbuettel) [inline R/utils-tiledb.R:33]: Should we refer to the constant `SOMA_LEGACY_VALIDITY_KEY` from utils?

File: R/utils-tiledb.R [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#' - the array contains the metadata tag `"soma_legacy_validity"` and it is set
+#'   to `"true"`
```
Comment (eddelbuettel) [inline R/utils-tiledb.R:35]: Should we make that a constant in utils.R ?

**Review by eddelbuettel — APPROVED** [review]
Looks very good. Three small 'editorial' questions inline.

File: DESCRIPTION (hunk 1 above)
Comment (aaronwolen) [reply DESCRIPTION:59]: Initially added in the hopes that pak would automatically pickup the dev version of tiledb-r from r-universe but no such luck. I'll drop it.

Comment (eddelbuettel) [reply DESCRIPTION:59]: Thanks for cleaning it up.  By my reading of Writing R Extensions, the field is only used for Suggests: so `pak` may be skipping it on purpose.

**Review by eddelbuettel — APPROVED** [review]
Looks good!

**Pull request merged**